src/buildSummary.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The MathWorks, Inc.
+import * as core from "@actions/core";
+import { join } from 'path';
+import { readFile } from 'fs';
+import { promisify } from 'util';
+
+// Promisify the readFile function to use it with async/await
+const readFileAsync = promisify(readFile);
+
+export interface Task {
+  name: string;
+  description: string;
+  failed: boolean;
+  skipped: boolean;
+  duration: string;
+}
+
+export interface TaskList {
+  taskDetails: Task[];
+}
+
+export async function readJsonFile(filePath: string): Promise<TaskList> {
+  try {
+    const data = await readFileAsync(filePath, { encoding: 'utf8' });
+    return JSON.parse(data);
+  } catch (error) {
+    console.error('Error reading the buildSummary file:', error);
+    throw error;
+  }
+}
+
+export function getBuildSummaryTable(tasks: TaskList): string[][] {
+  const header: string[] = ['MATLAB Build Task', 'Status', 'Description', 'Duration (HH:MM:SS)'];
+  let taskSummaryTableRows: string[][] = [header];
+
+  tasks.taskDetails.forEach((task, index) => {
```

Comment:
If we simplify the annotation and get rid of the `TaskList` wrapper type, this simplifies to
```
  tasks.forEach((task, index) => {
```

Reply:
Done.